contracts/deploy/017_ousd_upgrade.js

```diff
@@ -0,0 +1,60 @@
+const { isMainnet, isRinkeby } = require("../test/helpers.js");
+const { getTxOpts } = require("../utils/tx");
+
+let totalDeployGasUsed = 0;
+
+// Wait for 3 blocks confirmation on Mainnet/Rinkeby.
+const NUM_CONFIRMATIONS = isMainnet || isRinkeby ? 3 : 0;
+
+function log(msg, deployResult = null) {
+  if (isMainnet || isRinkeby || process.env.VERBOSE) {
+    if (deployResult) {
+      const gasUsed = Number(deployResult.receipt.gasUsed.toString());
+      totalDeployGasUsed += gasUsed;
+      msg += ` Address: ${deployResult.address} Gas Used: ${gasUsed}`;
+    }
+    console.log("INFO:", msg);
+  }
+}
+
+const upgradeOusd = async ({ getNamedAccounts, deployments }) => {
+  let transaction;
+
+  const { deploy } = deployments;
+  const { governorAddr, deployerAddr } = await getNamedAccounts();
+
+  console.log("Running 017_ousd_upgrade deployment...");
+
+  const sGovernor = ethers.provider.getSigner(governorAddr);
+
+  // Deploy a new vault.
+  const dOusd = await deploy("OUSD", {
+    from: deployerAddr,
+    ...(await getTxOpts()),
+  });
+  await ethers.provider.waitForTransaction(
+    dOusd.receipt.transactionHash,
+    NUM_CONFIRMATIONS
+  );
+  log("Deployed OUSD", dOusd);
+
+  // Update the proxy to use the new ousd contract.
```

Comment:
This won't go thru on mainnet since the governor is multi-sig. It needs to be done via governance.
Perhaps add a ```if isRinkeby``` statement.